rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketAccessItemController.java

```diff
@@ -86,6 +100,36 @@
         return new ResponseEntity<>(pagedResources, HttpStatus.OK);
     }
 
+    /**
+     * This DELETE method delete all workbasketAccessItems that correspond the given accessId.
+     *
+     * @param accessId which need remove his workbasketAccessItems.
+     * @return ResponseEntity if the user is not authorized.
+     * @throws NotAuthorizedException if the user is not authorized.
+     */
+    @DeleteMapping("{accessId}")
+    public ResponseEntity<Void> removeWorkbasketAccessItems(@PathVariable("accessId") String accessId)
+        throws NotAuthorizedException, InvalidArgumentException {
+        if (!isGroup(accessId)) {
+            List<WorkbasketAccessItemExtended> workbasketAccessItemList = workbasketService.createWorkbasketAccessItemExtendedQuery()
+                .accessIdIn(accessId)
+                .list();
+
+            if (workbasketAccessItemList != null && !workbasketAccessItemList.isEmpty()) {
+                workbasketService.deleteWorkbasketAccessItemsForAccessId(accessId);
+            }
+        } else {
+            throw new InvalidArgumentException(
+                accessId + " corresponding to a group, not a user. You just can remove access items for a user");
+        }
+
+        return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+    }
+
+    private boolean isGroup(String accessId) {
```

Comment:
That is good. But we can improve this still a bit and move the logix to the LdapClient. Then we would encapsulate all Ldap configuration there and the REST controller would not need to know about it. What do you think?

Reply:
You are right! It's better than all about ldap is on ldap client.
Thank you for your advice